### PR TITLE
test pipeline

### PR DESCRIPTION
**Pull request by marouaneamz**

### CODE REVIEW

File: tests/test_datasets/test_transforms/test_formatting.py [hunk 1]
```diff
@@ -130,3 +130,53 @@ def test_repr(self):
...
+            'scale_factor': 1.0,
+            'flip': False,
+            'img': mmcv.imread(img_path),
+            'task1': 1,
+            'task3': 3,
+        }
+
+        cfg = dict(
+            type='FormatMultiTaskLabels',
+            tasks=['task1', 'task2', 'task3']
```
Comment (piercus) [inline tests/test_datasets/test_transforms/test_formatting.py]: is it mandatory to have the list of tasks here ? Can we do without it ?

Comment (marouaneamz) [reply tests/test_datasets/test_transforms/test_formatting.py]: I can't find a way to retrieve the labels of the results without the names of the tasks

File: mmcls/datasets/transforms/formatting.py [hunk 2]
```diff
@@ -157,11 +157,9 @@ def transform(self, results: dict) -> dict:
                 'in the data pipeline or images have been loaded in ')
 
         data_sample = MultiTaskDataSample(tasks=self.tasks)
```
Comment (piercus) [inline mmcls/datasets/transforms/formatting.py]: This is strange to put a tasks keyword in the MultiTaskDataSample constructor
The only possible argument is `metainfo=`

Please remove the tasks options in the pipeline, let's make it simple for now

**Review by piercus — CHANGES_REQUESTED** [review]

File: mmcls/datasets/transforms/formatting.py [hunk 3]
```diff
@@ -156,12 +156,10 @@ def transform(self, results: dict) -> dict:
                 'please make sure `LoadImageFromFile` has been added '
                 'in the data pipeline or images have been loaded in ')
 
-        data_sample = MultiTaskDataSample(tasks=self.tasks)
-        gt_label = {}
-        for task in self.tasks:
-            if task in results:
-                gt_label[task] = to_tensor(results[task])
-        data_sample.set_gt_label(gt_label)
+        data_sample = MultiTaskDataSample(metainfo=self.tasks)
```
Comment (piercus) [inline mmcls/datasets/transforms/formatting.py]: Let's call it `self.metainfo`
```suggestion
        data_sample = MultiTaskDataSample(metainfo=self.metainfo)
```

File: tests/test_datasets/test_transforms/test_formatting.py [hunk 4]
```diff
@@ -130,3 +130,52 @@ def test_repr(self):
...
+        del data['gt_label']
+        with self.assertWarnsRegex(Warning, 'Cannot get "img"'):
+            results = transform(copy.deepcopy(data))
+            self.assertNotIn('gt_label', results['data_samples'])
+
+    def test_repr(self):
+        cfg = dict(type='PackClsInputs', meta_keys=['flip', 'img_shape'])
+        transform = TRANSFORMS.build(cfg)
+        self.assertEqual(
+            repr(transform), "PackClsInputs(meta_keys=['flip', 'img_shape'])")
```
Comment (piercus) [inline tests/test_datasets/test_transforms/test_formatting.py]: This is strange, why are you testing PackClsInputs ?

**Review by piercus — CHANGES_REQUESTED** [review]

**Review by piercus — APPROVED** [review]

**Pull request merged**